File: 백준/Silver/1459. 걷기/걷기.py
import sys
input = sys.stdin.readline

# 입력 처리
x, y, w, s = map(int, input().strip().split())

# 초기 값 설정
answer = float('inf')
total_distance = (x + y) * w

# x와 y의 홀짝 여부
is_even_x = x % 2
is_even_y = y % 2

# 짝짝 또는 홀홀인 경우
if (is_even_x and is_even_y) or (not is_even_x and not is_even_y):
    answer = min(
        answer,
        total_distance,
        max(x, y) * s,
        min(x, y) * s + (max(x, y) - min(x, y)) * w
    )
# 짝홀 또는 홀짝인 경우
else:
    answer = min(
        answer,
        total_distance,
        min(x, y) * s + (max(x, y) - min(x, y)) * w,
        (max(x, y) - 1) * s + w
    )

# 결과 출력
print(answer)

File: 백준/Silver/1459. 걷기/걷기.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const [x, y, w, s] = input[0].split(" ").map(Number)
const isEvenX = x % 2
const isEvenY = y % 2
let answer = Number.MAX_SAFE_INTEGER
const def = (x + y) * w
if ((isEvenX && isEvenY) || (!isEvenX && !isEvenY)) {
  // 짝 짝 , 홀 홀
  answer = Math.min(
    answer,
    def,
    Math.max(x, y) * s,
    Math.min(x, y) * s + (Math.max(x, y) - Math.min(x, y)) * w,
  )
} else {
  answer = Math.min(
    answer,
    def,
    Math.min(x, y) * s + (Math.max(x, y) - Math.min(x, y)) * w,
    (Math.max(x, y) - 1) * s + w,
  )
}
console.log(answer)